packages/ra-core/src/controller/details/useShowContext.tsx

```diff
@@ -1,35 +1,62 @@
 import { useContext } from 'react';
+import merge from 'lodash/merge';
+
 import { Record } from '../../types';
 import { ShowContext } from './ShowContext';
 import { ShowControllerProps } from './useShowController';
 
+/**
+ * Hook to read the show controller props from the ShowContext.
+ *
+ * Mostly used within a <ShowContext.Provider> (e.g. as a descendent of <Show>).
+ *
+ * But you can also use it without a <ShowContext.Provider>. In this case, it is up to you
+ * to pass all the necessary props.
+ *
+ * The given props will take precedence over context values.
+ *
+ * @typedef {Object} ShowControllerProps
+ *
+ * @returns {ShowControllerProps} create controller props
+ *
+ * @see useShowController for how it is filled
+ *
+ */
 export const useShowContext = <RecordType extends Record = Record>(
     props?: Partial<ShowControllerProps<RecordType>>
 ): Partial<ShowControllerProps<RecordType>> => {
-    // Can't find a way to specify the RecordType when CreateContext is declared
+    // Can't find a way to specify the RecordType when ShowContext is declared
     // @ts-ignore
     const context = useContext<ShowControllerProps<RecordType>>(ShowContext);
 
-    if (!context.resource) {
-        /**
-         * The element isn't inside a <ShowContext.Provider>
-         * To avoid breakage in that case, fallback to props
-         *
-         * @deprecated - to be removed in 4.0
-         */
-        if (process.env.NODE_ENV !== 'production') {
-            console.log(
-                "Show components must be used inside a <ShowContext.Provider>. Relying on props rather than context to get Show data and callbacks is deprecated and won't be supported in the next major version of react-admin."
-            );
-        }
-        // Necessary for actions (EditActions) which expect a data prop containing the record
-        // @deprecated - to be removed in 4.0d
-        return {
-            ...props,
-            record: props.record || props.data,
```

Comment:
You didn't port it to extractShowContextProps as in the Edit version. I assume this isn't required for ShowActions?